Task-WebRTC.md

```diff
@@ -136,11 +138,15 @@ number.
 
 # Signalling Channel Handover
 
-As soon as both clients have exchanged the required messages and the
-WebRTC `RTCPeerConnection` instance informs the user application that
-the peer-to-peer connection setup is complete, the user application
-SHOULD request that the client hands over the signalling channel to a
-dedicated data channel:
+As soon as the user application has created a WebRTC `RTCPeerConnection`
+instance which intends to use this task for its signalling, the user
+application SHOULD request that the client hands over the signalling
+channel to a dedicated data channel before *offer* or *answer* of the
```

Comment:
It's the other way around: A data channel must be created before offer/answer are generated. Otherwise, data channel SDP lines will not appear in the SDP blob and using a data channel will require an ICE restart.

You can test that using [this example](https://webrtc.github.io/samples/src/content/peerconnection/audio/): Click the `Call` button and then paste this into the console:

```js
((label) => {
    let dc = pc1.createDataChannel('test');
    dc.onopen = function(event) {
        console.log('Data channel', dc.label, '(', dc.id, ')', 'open');
    };
    dc.onbufferedamountlow = function(event) {
        console.log('Data channel', dc.label, '(', dc.id, ')', 'buffered amount low');
    };
    dc.onerror = function(event) {
        console.error('Data channel', dc.label, '(', dc.id, ')', 'error:', event);
    };
    dc.onclose = function(event) {
        console.log('Data channel', dc.label, '(', dc.id, ')', 'closed');
    };
    dc.onmessage = function(event) {
        var length = event.data.size || event.data.byteLength || event.data.length;
        console.info('Data channel', dc.label, '(', dc.id, ')', 'message size:', length,
            'content:\n' + event.data);
    };
    window.dc = dc;
})('test');
```

Check `dc.readyState`. It will never open.